File: solutions/0122_best-time-to-buy-and-sell-stock-ii.py
# Problem: https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii
# Runtime: 2 ms

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        return sum([max(prices[i + 1] - prices[i], 0) for i in range(len(prices) - 1)])

File: solutions/0122_best-time-to-buy-and-sell-stock-ii.rb
# Problem: https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii
# Runtime: 0 ms

# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    total = 0
    lastPrice = prices[0]
    prices.each do |p|
        total += p - lastPrice if p > lastPrice
        lastPrice = p
    end
    return total
end